### PR TITLE
Remove blpop FindFirst hop after wakeup

### PR DESCRIPTION
**Pull request by dranikpg**
Solves #1153 

Adds the following changes:
- NotifySuspended runs only once and stores the wakeup key in the tx shard's data
- BLPOP after waking up doesn't run FindFirst and takes the key directly from the wakeup args

### CODE REVIEW

File: src/server/transaction.cc [hunk 1]
```diff
@@ -1266,6 +1266,13 @@ bool Transaction::NotifySuspended(TxId committed_txid, ShardId sid) {
     return false;
   }
 
+  // Wake a transaction only once on the first notify.
+  // We don't care about preserving the strict order with multiple operations running on blocking
+  // keys in parallel, because the internal order is not observable from outside either way.
+  uint64_t default_txid = UINT64_MAX;
+  if (!notify_txid_.compare_exchange_weak(default_txid, committed_txid, memory_order_relaxed))
```
Comment (romange) [inline src/server/transaction.cc:1273]: compare_exchange_weak is an overkill for "run-once" semantics.
Also, I wonder if `notify_txid_` is needed after this change or it's been used only here

**Review by romange — COMMENTED** [review]
Can you also verify with ruby/sidekick test from our playground repo that everything is working well and nothing deadlocks/crashes?

File: src/server/transaction.cc (hunk 1 above)
Comment (dranikpg) [reply src/server/transaction.cc]: Indeed, I forgot to simplify FindFirst.

> compare_exchange_weak is an overkill for "run-once" semantics.

How else should I do run-once? 👀

Comment (romange) [reply src/server/transaction.cc]: for example,
```
if (var.fetch_add(1, relaxed) == 0) {

}
```

**Comment by dranikpg** [comment]
I checked the ruby playground and it seems to work (worker rb + loadtest)

**Pull request merged**